homeassistant/components/kodi/browse_media.py

```diff
@@ -0,0 +1,214 @@
+                    "properties": ["thumbnail", "season", "tvshowid"],
+                }
+            )
+            media = media.get("seasons")
+            tvshow = await media_library._server.VideoLibrary.GetTVShowDetails(
+                {"tvshowid": int(search_id), "properties": ["thumbnail"]}
+            )
+            thumbnail = media_library.thumbnail_url(
+                tvshow["tvshowdetails"].get("thumbnail")
+            )
+            title = tvshow["tvshowdetails"]["label"]
+        else:
+            media = await media_library._server.VideoLibrary.GetTVShows(query)
+            media = media.get("tvshows")
+            title = "TV Shows"
+    elif search_type == MEDIA_TYPE_SEASON:
+        tv_show_id, season_id = search_id.split("/", 1)
+        media = await media_library._server.VideoLibrary.GetEpisodes(
+            {
+                "tvshowid": int(tv_show_id),
+                "season": int(season_id),
+                "properties": ["thumbnail", "tvshowid", "seasonid"],
+            }
+        )
+        media = media.get("episodes")
+        if media:
+            season = await media_library._server.VideoLibrary.GetSeasonDetails(
+                {"seasonid": int(media[0]["seasonid"]), "properties": ["thumbnail"]}
+            )
+            thumbnail = media_library.thumbnail_url(
+                season["seasondetails"].get("thumbnail")
+            )
+            title = season["seasondetails"]["label"]
+
+    if media is None:
+        return
+
+    return BrowseMedia(
+        media_content_id=payload["search_id"],
+        media_content_type=search_type,
+        title=title,
+        can_play=search_type in PLAYABLE_MEDIA_TYPES and search_id,
+        can_expand=True,
+        children=[item_payload(item, media_library) for item in media],
+        thumbnail=thumbnail,
+    )
+
+
+def item_payload(item, media_library):
+    """
+    Create response payload for a single media item.
+
+    Used by async_browse_media.
+    """
+    if "songid" in item:
+        media_content_type = MEDIA_TYPE_TRACK
+        media_content_id = f"{item['songid']}"
+    elif "albumid" in item:
+        media_content_type = MEDIA_TYPE_ALBUM
+        media_content_id = f"{item['albumid']}"
+    elif "artistid" in item:
+        media_content_type = MEDIA_TYPE_ARTIST
+        media_content_id = f"{item['artistid']}"
+    elif "movieid" in item:
+        media_content_type = MEDIA_TYPE_MOVIE
+        media_content_id = f"{item['movieid']}"
+    elif "episodeid" in item:
+        media_content_type = MEDIA_TYPE_EPISODE
+        media_content_id = f"{item['episodeid']}"
+    elif "seasonid" in item:
+        media_content_type = MEDIA_TYPE_SEASON
+        media_content_id = f"{item['tvshowid']}/{item['season']}"
+    elif "tvshowid" in item:
+        media_content_type = MEDIA_TYPE_TVSHOW
+        media_content_id = f"{item['tvshowid']}"
+    else:
+        media_content_type = item.get("type")
```

Comment:
Please make a code comment about all the possible content types that this can return.